### PR TITLE
Fat 5517 create filterforproducts

### PR DESCRIPTION
**Pull request by Activict**

### CODE REVIEW

File: StoreApp.BLL/Services/FilterProductsService.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        {
+            DataBase = uof;
+            config = new MapperConfiguration(cfg => { cfg.CreateMap<Product, ProductDTO>(); }).CreateMapper();
+        }
+
+        public IEnumerable<ProductDTO> GetProductsDTOFiltered(FilterProductsDTO filter)
+        {
+            var products = config.Map<IEnumerable<Product>, IEnumerable<ProductDTO>>(DataBase.Products.GetAll());
+
+            if (filter.Name != null)
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/FilterProductsService.cs]: You can use standard methods of the class String
For example: **String.IsNullOrEmpty()**

File: StoreApp.BLL/Services/FilterProductsService.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+            if (filter.PriceTo > 0)
+            {
+                products = products.Where(p => p.Price < filter.PriceTo);
+            }
+
+            if (filter.Enable == true)
+            {
+                products = products.Where(p => p.Enable == true);
+            }
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/FilterProductsService.cs]: Just use a simple condition, for example, **if(filter.Enable)**
Also pay attention to the record 49

File: StoreApp.BLL/Services/FilterProductsService.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+using System.Collections.Generic;
+using System.Linq;
+
+namespace StoreApp.BLL.Services
+{
+    public class FilterProductsService
+    {
+        private IUnitOfWork DataBase { get; set; }
+
+        private IMapper config;
```
Comment (AleksandroBoko) [inline StoreApp.BLL/Services/FilterProductsService.cs]: We skipped this point but I guess it's better to keep this property and field together. Also we should follow one common style - declare properties or fields. 
And look at the difference between styles of naming.

File: StoreApp.BLL/DTO/FilterProductsDTO.cs [hunk 4]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (AleksandroBoko) [inline StoreApp.BLL/DTO/FilterProductsDTO.cs]: Do you need to keep all these namespaces?

File: StoreApp/Controllers/StoreController.cs [hunk 5]
```diff
@@ -39,6 +43,34 @@ public ActionResult Index()
...
+            }
+            
+            FilterProductsDTO filterDTO = webMapper.config.Map<FilterProductsViewModel, FilterProductsDTO>(filter);
+            
+            var productsFiltered = filterProductsService.GetProductsDTOFiltered(filterDTO).Select(p => webMapper.Map(p));
+
+            return View(productsFiltered);
+        }
+
+        public ActionResult FilterMenuPartial()
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs:66]: If you mark all methods by using attributes, don't skip this one

File: StoreApp/Controllers/StoreController.cs [hunk 6]
```diff
@@ -39,6 +43,34 @@ public ActionResult Index()
...
+        {
+            TempData["Categories"] = new SelectList(categoryService.GetAll(), dataValueField: "Id", dataTextField: "Name");
+            TempData["Brands"] = new SelectList(brandService.GetAll(), dataValueField: "Id", dataTextField: "Name");
+            TempData["Producers"] = new SelectList(producerService.GetAll(), dataValueField: "Id", dataTextField: "Name");
```
Comment (AleksandroBoko) [inline StoreApp/Controllers/StoreController.cs]: As I can see we use this block of code a lot of times. So I propose to optimize code and implement the principle DRY.

File: StoreApp/Util/WebMapper.cs [hunk 7]
```diff
@@ -24,6 +25,7 @@ public WebMapper()
                 cfg.CreateMap<ProductDTO, EditProductViewModel>();
                 cfg.CreateMap<EditProductViewModel, ProductDTO>();
                 cfg.CreateMap<ProductDTO, ProductViewModel>();
+                cfg.CreateMap<FilterProductsViewModel, FilterProductsDTO>();
```
Comment (AleksandroBoko) [inline StoreApp/Util/WebMapper.cs]: I propose such format
http://prntscr.com/qc74l4

File: StoreApp.BLL/Services/FilterProductsService.cs (hunk 1 above)
Comment (Activict) [reply StoreApp.BLL/Services/FilterProductsService.cs]: fixed

File: StoreApp.BLL/Services/FilterProductsService.cs (hunk 2 above)
Comment (Activict) [reply StoreApp.BLL/Services/FilterProductsService.cs]: fixed

File: StoreApp.BLL/Services/FilterProductsService.cs (hunk 3 above)
Comment (Activict) [reply StoreApp.BLL/Services/FilterProductsService.cs]: fixed

File: StoreApp.BLL/DTO/FilterProductsDTO.cs (hunk 4 above)
Comment (Activict) [reply StoreApp.BLL/DTO/FilterProductsDTO.cs]: fixed

File: StoreApp/Controllers/StoreController.cs (hunk 5 above)
Comment (Activict) [reply StoreApp/Controllers/StoreController.cs:66]: fixed

File: StoreApp/Util/WebMapper.cs (hunk 7 above)
Comment (Activict) [reply StoreApp/Util/WebMapper.cs]: fixed

**Pull request merged**